packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -254,26 +261,29 @@ class FlutterDevice {
       await device.stopApp(package);
       return;
     }
-    final List<FlutterView> flutterViews = views;
-    if (flutterViews == null || flutterViews.isEmpty) {
+    await refreshViews();
+    if (views == null || views.isEmpty) {
       return;
     }
     // If any of the flutter views are paused, we might not be able to
     // cleanly exit since the service extension may not have been registered.
-    if (flutterViews.any((FlutterView view) {
-      return view != null &&
-             view.uiIsolate != null &&
-             view.uiIsolate.pauseEvent != null &&
-             view.uiIsolate.pauseEvent.isPauseEvent;
+    for (final FlutterView flutterView in views) {
+      final vm_service.Isolate isolate = await vmService
+        .getIsolate(flutterView.uiIsolate.id)
+        .catchError((dynamic error, StackTrace stackTrace) {
```

Comment:
I added a catchError field to all of these because it seems like that is what we were doing before:


https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/vmservice.dart#L1035

Though before we might have been re-using invalid isolates, which is probably worse

Reply:
Would a trace message here be useful?

Reply:
Also, this pattern is repeated several times, so creating a new method somewhere for this could be a good cleanup.

Reply:
The previous trace was not useful, since we don't seem to hit this case often.

Updated to an extension method on VmService